From 82d600055f5722f33c3994b1e4e6869fc94f5937 Mon Sep 17 00:00:00 2001
From: Dong-Heon Jung <dongheon.jung@lge.com>
Date: Tue, 11 Nov 2014 16:37:00 +0900
Subject: [PATCH] QNetworkDiskCache: Disable disk cache if content have cookie
 informations

If we save cookie information to disk cache and load from disk cache,
it can cause some security issues.
So we disable disk cache for contents which have cookie information

Change-Id: Iacae9938046bc19d1de0b043bfadeab7b2a9489a
Upstream-Status: unknown
---
 src/network/access/qnetworkreplyhttpimpl.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/network/access/qnetworkreplyhttpimpl.cpp b/src/network/access/qnetworkreplyhttpimpl.cpp
index a0c2214a50..dd86015215 100644
--- a/src/network/access/qnetworkreplyhttpimpl.cpp
+++ b/src/network/access/qnetworkreplyhttpimpl.cpp
@@ -1744,6 +1744,11 @@ QNetworkCacheMetaData QNetworkReplyHttpImplPrivate::fetchCacheMetaData(const QNe
         else if (cacheControl.contains("no-store"))
             canDiskCache = false;
 
+        // Disable disk cache for contents which has set-cookie
+        it = cacheHeaders.findRawHeader("set-cookie");
+        if (it != cacheHeaders.rawHeaders.constEnd())
+            canDiskCache = false;
+
         // Check the Content-Type header.
         // If type is video or audio, disables disk cache.
         it = cacheHeaders.findRawHeader("content-type");
